dipy/reconst/ivim.py

```diff
@@ -0,0 +1,391 @@
+
+        S_0, f, D* and D are the IVIM parameters.
+
+    Parameters
+    ----------
+    params : array
+        An array of IVIM parameters - S0, f, D_star, D
+
+    bvals : array
+        bvalues
+
+    References
+    ----------
+    .. [1] Le Bihan, Denis, et al. "Separation of diffusion
+               and perfusion in intravoxel incoherent motion MR
+               imaging." Radiology 168.2 (1988): 497-505.
+    .. [2] Federau, Christian, et al. "Quantitative measurement
+               of brain perfusion with intravoxel incoherent motion
+               MR imaging." Radiology 265.3 (2012): 874-881.
+    """
+    S0, f, D_star, D = params
+    S = S0 * (f * np.exp(-bvals * D_star) + (1 - f) * np.exp(-bvals * D))
+    return S
+
+
+def _ivim_error(params, bvals, signal):
+    """Error function to be used in fitting the IVIM model
+
+    Parameters
+    ----------
+    params : array
+        An array of IVIM parameters. [S0, f, D_star, D]
+
+    bvals : array
+        bvalues
+
+    signal : array
+        Array containing the actual signal values.
+
+    """
+    return (signal - ivim_function(params, bvals))
+
+
+class IvimModel(ReconstModel):
+    """Ivim model
+    """
+
+    def __init__(self, gtab, split_b=200.0,
+                 bounds=None, tol=1e-7,
+                 options={'gtol': 1e-7, 'ftol': 1e-7,
+                          'eps': 1e-7, 'maxiter': 1000}):
+        """
+        Initialize an IVIM model.
+
+        The IVIM model assumes that biological tissue includes a volume
+        fraction 'f' of water flowing in perfused capillaries, with a
+        perfusion coefficient D* and a fraction (1-f) of static (diffusion
+        only), intra and extracellular water, with a diffusion coefficient
+        D. In this model the echo attenuation of a signal in a single voxel
+        can be written as
+
+            .. math::
+
+            S(b) = S_0[f*e^{(-b*D\*)} + (1-f)e^{(-b*D)}]
+
+            Where:
+            .. math::
+
+            S_0, f, D\* and D are the IVIM parameters.
+
+        Parameters
+        ----------
+        gtab : GradientTable class instance
+            Gradient directions and bvalues
+
+        split_b : float
+            The b-value to split the data on for two-stage fit
+
+        bounds : array, (4,4)
+            Bounds for the parameters. This is applicable only for Scipy
+            version > 0.17 as we use least_squares for the fitting which
+            supports bounds. For versions less than Scipy 0.17, this is
+            by default set to None. Setting a bound on a Scipy version less
+            than 0.17 will raise an error.
+
+            This can be supplied as a tuple for Scipy versions 0.17. It is
+            recommended to upgrade to Scipy 0.17 for bounded fitting. The
+            default bounds are set to ([0., 0., 0., 0.], [np.inf, 1., 1., 1.])
+
+        tol : float, optional
+            Default : 1e-7
+            Tolerance for convergence of minimization.
```

Comment:
Explanation first then Default - keeping structure consistent is the way of PEP8 :)